src/base/net/downloadmanager.h

```diff
@@ -42,16 +45,40 @@ namespace Net
 {
     class DownloadHandler;
 
+    struct DownloadRequest
+    {
+        DownloadRequest(const QString &url);
```

Comment:
>From the other hand, eliminating it would make it possible to use aggregate initialisation.

It was my initial idea. Unfortunately, we can't use aggregate initiali*z*ation until c++14 if some field has default value.
Since in most cases we need only URL (all other fields keep its default values), I added this constructor.

Reply:
I found only a single place where an instance of the class is created, namely in `Net::DownloadManager::downloadUrl()`, and there almost all the other fields are set after the constructor call. Thus, sorry, I can only repeat myself: what's the purpose of the constructor?

Reply:
If you can't look a little forward, I'll explain. The new `DownloadManager::download()` method is planned as a replacement for the legacy `DownloadManager::downloadUrl()` and in most use cases only URL is used to download something. I just don't want to change a lot of code in this PR. Its main purpose is different.

Reply:
Could you do me a favour leading me a little bit further to show how does this constructor provide better experience as comparing to other considered options, especially when not only an URL needs to be passed in? 

> in most use cases only URL is used to download something.

I'm also interested to learn how do 2 out of 15 (in the qBt source base)  mutate into the "most use cases".

Reply:
Well, you made me look into the future again and rethink it...
First. All "optional" parts of download request have zero-equal defaults so I can omit it in declaration and use aggregate initialization.
Second. Why do I need this structure (apart from simpler and more scalable management inside the implementation)? The premise is that there is only one required parameter when sending a request - the URL. All others are optional and most of them usually uses its default value. So I just wanted to prevent them from being explicitly mentioned when sending the request. Unfortunately they are used in arbitrary order. So we can't omit params before the one we need to set explicitly, e.g. when we need to set user agent (at least untill c++20).

Apparently, I will have to use the approach used to build SQL queries in some frameworks, e.g.:
```c++
DownloadManager::download(DownloadRequest().url("http://some-url").userAgent("qBittorrent"));
```

Reply:
FYI, it's called "named parameters idiom".